File: solutions/128.longest-consecutive-sequence.java
/*
 * @lc app=leetcode id=128 lang=java
 *
 * [128] Longest Consecutive Sequence
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> nset = new HashSet<>(nums.length);
        for (int i : nums) {
            nset.add(i);
        }

        int maxSize = 0;
        for (int i : nums) {
            if (!nset.contains(i-1)) {
                int last = i+1;
                while (nset.contains(last)) {
                    last++;
                }
                int size = last - i;
                maxSize = Math.max(size, maxSize);
            }
            if (maxSize > nums.length / 2) break;
        }
        return maxSize;
    }
}
// @lc code=end

File: solutions/128.longest-consecutive-sequence.py
#
# @lc app=leetcode id=128 lang=python3
#
# [128] Longest Consecutive Sequence
#

# @lc code=start

from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nset = set(nums)
        maxSize = 0
        for i in nset:
            if (i-1) not in nset:
                last = i+1
                while last in nset:
                    last += 1
                size = last - i
                maxSize = max(maxSize, size)

            if maxSize > len(nums) // 2:
                break

        return maxSize

# @lc code=end

